crates/store/re_chunk_store/src/query.rs

```diff
@@ -413,3 +560,201 @@ impl ChunkStore {
             .collect()
     }
 }
+
+// Queries returning `usize` and `bool`
+impl ChunkStore {
+    /// Returns the number of events logged for a given component on the given entity path across all timelines.
+    pub fn num_events_on_timeline_for_component(
+        &self,
+        timeline: &Timeline,
+        entity_path: &EntityPath,
+        component_name: ComponentName,
+    ) -> usize {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        self.num_events_for_static_component(entity_path, component_name)
+            + self.num_events_on_timeline_for_temporal_component(
+                timeline,
+                entity_path,
+                component_name,
+            )
+    }
+
+    /// Returns the min and max times at which data was logged for an entity
+    /// on a specific timeline.
+    pub fn time_range_for_entity_on_timeline(
+        &self,
+        timeline: &Timeline,
+        entity_path: &EntityPath,
+    ) -> Option<ResolvedTimeRange> {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        let temporal_chunk_ids_per_timeline =
+            self.temporal_chunk_ids_per_entity.get(entity_path)?;
+        let chunk_id_sets = temporal_chunk_ids_per_timeline.get(timeline)?;
+
+        let start = chunk_id_sets.per_start_time.first_key_value()?.0;
+        let end = chunk_id_sets.per_end_time.last_key_value()?.0;
+
+        Some(ResolvedTimeRange::new(*start, *end))
+    }
+
+    /// Returns the number of events logged for an entity on a specific timeline.
```

Comment:
How does this behave wrt static data?

Reply:
You're right, in this case we actually don't want that